extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -362,4 +373,50 @@ private synchronized void setPatchMatchingPolicyEnabled() {
 
         protected abstract boolean httpPermissionsEmpty();
     }
+
+    public void setMtlsCertificateRoleProperties(HttpConfiguration config) {
+        if (config.auth.certificateRoleProperties.isPresent()) {
+            Map<String, Set<String>> roles = new HashMap<>();
+
+            Path rolesPath = config.auth.certificateRoleProperties.get();
+            URL rolesResource = null;
+            if (Files.exists(rolesPath)) {
+                try {
+                    rolesResource = rolesPath.toUri().toURL();
+                } catch (MalformedURLException e) {
+                    // The Files.exists(rolesPath) check has succeeded therefore this exception can't happen in this case
+                }
+            } else {
+                rolesResource = Thread.currentThread().getContextClassLoader().getResource(rolesPath.toString());
+            }
+            if (rolesResource == null) {
+                throw new ConfigurationException(
+                        "quarkus.http.auth.certificate-role-properties location can not be resolved",
+                        Set.of("quarkus.http.auth.certificate-role-properties"));
+            }
+            Properties rolesProps = new Properties();
+            try (InputStream is = rolesResource.openStream()) {
+                rolesProps.load(is);
```

Comment:
Whenever we read a properties file from the user, we should always wrap it with an `InputStreamReader` using `StandardCharsets.UTF_8` for the charset and load it from that using the `Properties#load(Reader)` overload.

It might also be prudent to add a `BufferedReader` and/or `BufferedInputStream` layer.